### ISSUE TITLE
Include PeerTube (Sepia Search) to

### ISSUE THREAD

**Issue by KubikPixel** [pystardust/ytfzf#59]
Please include in the search result also videos from _Sepia Search_ (_PeerTube_) as a alternative way to find great **FLOSS** Videos.

- [Sepia Search](https://sepiasearch.org/)
- [PeerTube](https://joinpeertube.org/)

**Comment by pystardust** [comment]
Sounds interesting, just found out that you can play peertube videos with mpv and ytdl. I will need to go throught the peertube api.
Does anybody know on how to get peertube api? Or we can scrape the site just like youtube.

**Comment by cybarspace** [comment]
> Does anybody know on how to get peertube api? Or we can scrape the site just like youtube.

I think we can just scrape it.

That being said I'd also like support for:
- [LBRY](https://lbry.tv/)
- [Odysee](https://odysee.com/)

Maybe there can be a flag that brings up a menu that lists different video sources like YT, LBRY, etc. and when the user selects one of them, it presents a prompt where user enters search string and presses `Enter` and then `ytfzf` will behave just as it is now, but for the chosen site.

**Comment by hp77-creator** [comment]
Yea flag will keep the logic modular for diff. websites. and also break less as new sites are added.

**Comment by pystardust** [comment]
Noted. We will add this to the TODO's and add sites one by one

**Comment by Toasterbirb** [comment]
Also invidious would be nice to be supported as an alternative source for youtube videos. It should work without javascript as well, so getting the results should be fairly simple.

**Comment by pystardust** [comment]
The current script relies on the json file from youtube's search result page.
A browser would run the javascript and render the search results.
`ytfzf` scrapes the contents of the java script's json, and instead of running any java script, it uses `jq` to parse the json object and render the video results.

**Comment by qoheniac** [comment]
Actually, Sepia Search has a JSON API that can be used like this:

https://sepiasearch.org/api/v1/search/videos?search=stallman

Not sure where this is documented, but I found it [in NewPipe’s code](https://github.com/TeamNewPipe/NewPipeExtractor/blob/564d74c250590add5ecf6f604c7cb0c090b23b79/extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/linkHandler/PeertubeSearchQueryHandlerFactory.java).

**Comment by qoheniac** [comment]
I just had a few minutes to think about this an wrote a few lines to produce `videos_json` data like in Ytfzf's code but for Sepia Search instead of YouTube.

```sh
# query string for testing
search_query="Hello World"

scrape_pt() {
     #gets a list of videos
     videos_json=$(
     curl \
         -s "https://sepiasearch.org/api/v1/search/videos" \
         -G --data-urlencode "search=$*" |
     jq '[ .data | .[] |
         {
              title: .name,
              channel: .channel.displayName,
              duration: .duration,
              views: .views,
              date: .publishedAt,
              videoID: .uuid,
              thumbs: .thumbnailUrl
          }
     ]')
}

scrape_pt "$search_query"

# print first three entries
printf "%s" "$videos_json" | jq '[.[]][:3]'
```

The output of the above looks like this:
```json
[
  {
    "title": "Hello World",
    "channel": "Main requiem channel",
    "duration": 58,
    "views": 7,
    "date": "2019-12-30T18:12:19.662Z",
    "videoID": "e3fbacb5-2add-4f45-8bec-7ee565dddc8f",
    "thumbs": "https://video.hackers.town/static/thumbnails/e3fbacb5-2add-4f45-8bec-7ee565dddc8f.jpg"
  },
  {
    "title": "Hello World",
    "channel": "Main programmingfish channel",
    "duration": 6,
    "views": 16,
    "date": "2019-07-07T13:26:01.663Z",
    "videoID": "7cf70066-7775-4d2f-a9c2-44c83fd73955",
    "thumbs": "https://tube.piweb.be/static/thumbnails/7cf70066-7775-4d2f-a9c2-44c83fd73955.jpg"
  },
  {
    "title": "4- Hello World",
    "channel": "LearnAsYouLike",
    "duration": 325,
    "views": 0,
    "date": "2020-11-15T04:00:21.357Z",
    "videoID": "a2576ad6-1efb-4f06-917b-b32bc3f44fdf",
    "thumbs": "https://diode.zone/static/thumbnails/a2576ad6-1efb-4f06-917b-b32bc3f44fdf.jpg"
  }
]
```

**Comment by Euro20179** [comment]
wow, nice, I will be trying to implement this.

**Comment by pystardust** [comment]
Thanks to @Euro20179 this has been implemented in #281 (not yet merged to master,
to use development branch `ytfzf --update-unstable`)
Search peertube with `-P` flag
